lib/core-server/src/core-presets.test.ts

```diff
@@ -135,44 +141,35 @@ describe.each([
   ['web-components-kitchen-sink', webComponentsOptions],
   ['html-kitchen-sink', htmlOptions],
 ])('%s', (example, frameworkOptions) => {
-  beforeEach(() => {
-    jest.clearAllMocks();
-    cache.clear();
-  });
-
-  const options = {
-    ...baseOptions,
-    ...frameworkOptions,
-    configDir: path.resolve(`${__dirname}/../../../examples/${example}/.storybook`),
-  };
-
-  describe('manager', () => {
-    it('dev mode', async () => {
-      await buildDevStandalone({ ...options, ignorePreview: true });
-
-      const managerConfig = prepareSnap(managerExecutor.get, 'manager');
-      expect(managerConfig).toMatchSpecificSnapshot(snap(`${example}_manager-dev`));
-    });
-    it('production mode', async () => {
-      await buildStaticStandalone({ ...options, ignorePreview: true });
-
-      const managerConfig = prepareSnap(managerExecutor.get, 'manager');
-      expect(managerConfig).toMatchSpecificSnapshot(snap(`${example}_manager-prod`));
-    });
-  });
-
-  describe('preview', () => {
-    it('dev mode', async () => {
-      await buildDevStandalone({ ...options, managerCache: true });
-
-      const previewConfig = prepareSnap(previewExecutor.get, 'preview');
-      expect(previewConfig).toMatchSpecificSnapshot(snap(`${example}_preview-dev`));
+  describe.each([
+    ['manager', managerExecutor],
+    ['preview', previewExecutor],
+  ])('%s', (component, executor) => {
+    beforeEach(async () => {
+      jest.clearAllMocks();
+      await cache.clear();
```

Comment:
This test appeared to be quite flakey on mac and windows, seemingly due to tests stepping on each other, even though they run in sequence.  The changes I've made here do two main things:

- use nested `.each` blocks so that we can DRY it up a bit
- use a temporary folder to build storybook into, rather than building into the same directory in the repo each time and relying on it being cleaned up properly between tests.  I believe that is what was causing the flake, because after I changed to use a temp folder, those problems went away.